src/services/item/plugins/html/validator.ts

```diff
@@ -0,0 +1,18 @@
+import { H5P } from './h5p/validation/h5p';
+
+export interface HtmlValidator {
+  /**
+   * Checks whether a given file extension is allowed inside a package
+   * @param extension A string representing the file extension (may or may not contain leading dot or be uppercase)
+   * @return true if the file extension is allowed, false otherwise
+   */
+  isExtensionAllowed(extension: string): boolean;
+
+  /**
+   * Validates an extracted html package content
+   */
+  validatePackage(
+    extractedRoot: string,
+    // TODO: type
+  ): Promise<{ isValid: false; error: string } | { isValid: true; manifest?: H5P.Manifest }>;
```

Comment:
Yeah, we can simplify into a `Promise<T>` where T is generic from the interface depending on the concrete html service type, and instead of returning `isValid, error` we should simply throw from within.